src/core/stream_recv.c

```diff
@@ -328,6 +328,11 @@ QuicStreamProcessStreamFrame(
         Status = QUIC_STATUS_SUCCESS;
 
     } else {
+        uint64_t MaxData = Stream->Connection->Send.MaxData;
+        if (EncryptedWith0Rtt) {
+            QUIC_FRE_ASSERT(Stream->Connection->ResumedTP);
+            MaxData = Stream->Connection->ResumedTP->InitialMaxData;
+        }
```

Comment:
This logic isn't correct I think, as the connection's MaxData is the current value, not the initial, but you're exchanging it with the 0-RTT initial.